File: app/src/main/java/org/autojs/autojs/ui/splash/SplashActivity.java
package org.autojs.autojs.ui.splash;

import static android.view.View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN;
import static android.view.View.SYSTEM_UI_FLAG_LAYOUT_STABLE;
import static android.view.WindowManager.LayoutParams.FLAG_LAYOUT_NO_LIMITS;

import android.os.Bundle;
import android.os.Handler;

import androidx.annotation.Nullable;

import org.autojs.autojs.app.tool.FloatingButtonTool;
import org.autojs.autojs.ui.BaseActivity;
import org.autojs.autojs.ui.floating.FloatyWindowManger;
import org.autojs.autojs.ui.main.MainActivity_;
import org.autojs.autojs.util.ViewUtils;
import org.autojs.autojs6.R;

/**
 * Created by Stardust on 2017/7/7.
 */
public class SplashActivity extends BaseActivity {

    private static final long INIT_TIMEOUT = 700;

    private boolean mAlreadyEnterNextActivity = false;
    private boolean mPaused;
    private Handler mHandler;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        init();
        mHandler.postDelayed(SplashActivity.this::enterNextActivity, INIT_TIMEOUT);
        if (FloatyWindowManger.isCircularMenuShowing()) {
            FloatyWindowManger.hideCircularMenu();
        }
        super.onCreate(savedInstanceState);
    }

    private void init() {
        setContentView(R.layout.activity_splash);
        mHandler = new Handler();
    }

    @Override
    protected void onPause() {
        super.onPause();
        mPaused = true;
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (mPaused) {
            mPaused = false;
            enterNextActivity();
        }
    }

    void enterNextActivity() {
        if (mAlreadyEnterNextActivity)
            return;
        if (mPaused) {
            return;
        }
        mAlreadyEnterNextActivity = true;
        MainActivity_.intent(this).start();
        finish();
    }

}

File: app/src/main/java/org/autojs/autojs/ui/splash/SplashActivity.kt
package org.autojs.autojs.ui.splash

import android.annotation.SuppressLint
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import org.autojs.autojs.ui.BaseActivity
import org.autojs.autojs.ui.floating.FloatyWindowManger
import org.autojs.autojs.ui.main.MainActivity
import org.autojs.autojs6.databinding.ActivitySplashBinding

/**
 * Created by Stardust on Jul 7, 2017.
 * Modified by SuperMonster003 as of Apr 4, 2023.
 * Transformed by SuperMonster003 on Apr 4, 2023.
 */
@SuppressLint("CustomSplashScreen")
class SplashActivity : BaseActivity() {

    override val handleStatusBarThemeColorAutomatically = false

    private var mAlreadyEnterNextActivity = false
    private var mPaused = false

    private lateinit var mHandler: Handler

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(ActivitySplashBinding.inflate(layoutInflater).root)

        @Suppress("DEPRECATION")
        mHandler = Looper.myLooper()?.let { Handler(it) } ?: Handler()
        mHandler.postDelayed({ enterNextActivity() }, INIT_TIMEOUT)

        FloatyWindowManger.hideCircularMenuIfNeeded()
    }

    override fun onPause() {
        super.onPause()
        mPaused = true
    }

    override fun onResume() {
        super.onResume()
        if (mPaused) {
            mPaused = false
            enterNextActivity()
        }
    }

    private fun enterNextActivity() {
        if (!mAlreadyEnterNextActivity && !mPaused) {
            mAlreadyEnterNextActivity = true
            MainActivity.launch(this)
            finish()
        }
    }

    companion object {

        const val INIT_TIMEOUT = 1000L

    }

}